src/ffi/macros.rs

```diff
@@ -17,7 +17,43 @@ macro_rules! ffi_fn {
         }
     };
 
+    ($(#[$doc:meta])* fn $name:ident($($arg:ident: $arg_ty:ty),*) -> $ret:ty $body:block ?= $default:expr) => {
```

Comment:
I think it could be possible to not need to repeat ourselves, by having the the `?=` the top variant, and have a macro rule variant that doesn't match `?=`, and those would just call the macro again with `ffi_fn!($(#[$doc])* fn $name($($arg: $arg_ty),*) -> $ret $body ?= crate::ffi::ffi_panic());` or so, kinda like the others that have the return type auto-added. Right?

Reply:
I'll fix that. Do you want us to still eprint before a panic like before?

Reply:
Yea, probably worth still printing before the abort. Though, now that I see it's just an expression, it probably doesn't need to be an extra function call, just a block defined in the macro.